chromium_src/components/bookmarks/browser/bookmark_model_unittest.cc

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "components/bookmarks/test/test_bookmark_client.h"
+
+namespace bookmarks {
+class BraveTestBookmarkClient : public TestBookmarkClient {
+ public:
+  BraveTestBookmarkClient() : TestBookmarkClient() {}
+  ~BraveTestBookmarkClient() override {}
+
+  static std::unique_ptr<BookmarkModel> CreateModel() {
+  return CreateModelWithClient(base::WrapUnique(new BraveTestBookmarkClient));
+  }
+
+ private:
+  // BookmarkClient:
+  bool IsPermanentNodeVisible(const BookmarkPermanentNode* node) override {
```

Comment:
It's not clear to me what we're doing here. Changing this in a test-only bookmark client doesn't seem to really validate anything useful for us in the test?

Reply:
addressed in 0f076bb8ef52f06fa76ddaecf2b2760cf6f577ca